modules/signals/src/state-source.ts

```diff
@@ -49,6 +53,14 @@ export function patchState<State extends object>(
   notifyWatchers(stateSource);
 }
 
+function freezeInDevMode<State extends object>(value: State): State {
+  if (ngDevMode) {
+    Object.freeze(value);
```

Comment:
To ensure that nested state properties are also not mutated, we can deeply freeze the state object.

Reply:
I've copied and slightly modified the existing freeze function of the global store.